src/compas_viewer/components/dialog.py

```diff
@@ -0,0 +1,96 @@
+from PySide6.QtWidgets import QDialog
+from PySide6.QtWidgets import QHBoxLayout
+from PySide6.QtWidgets import QLabel
+from PySide6.QtWidgets import QPushButton
+from PySide6.QtWidgets import QVBoxLayout
+
+from compas_viewer.base import Base
+from compas_viewer.components.double_edit import DoubleEdit
+
+
+class CameraSettingsDialog(QDialog, Base):
+    """
+    Dialog for adjusting camera settings in a graphical user interface.
+
+    This dialog allows users to dynamically modify the camera's position and target coordinates
+    through the use of spin boxes for each coordinate axis (X, Y, Z).
+
+    Attributes
+    ----------
+    layout : QVBoxLayout
+        The layout for arranging widgets vertically within the dialog.
+    spin_boxes : dict
+        Stores references to spin box widgets for easy access when updating camera settings.
+    update_button : QPushButton
+        Button for applying the updated camera settings and closing the dialog.
+
+    Methods
+    -------
+    __init__(self):
+        Initializes the dialog with spin boxes for camera position and target settings.
+    updateCameraTarget(self):
+        Updates the camera's position and target based on the spin box values and closes the dialog.
```

Comment:
remove this. replace by a "Parameters" section. see other classes in COMPAS for reference...